File: LeetCode c#/145-BinaryTreePostOrderTraversal.cs
// 145. Binary Tree Postorder Traversal

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public IList<int> PostorderTraversal(TreeNode root) {
        var oList = new List<int>();
        
        TT(root, oList);
        
        return oList;
    }
    
    void TT(TreeNode node, List<int> oList) {
        if (node == null) return;
        
        TT(node.left, oList);
        TT(node.right, oList);
        
        oList.Add(node.val);
        
        
    }
}

File: LeetCode Java/Data Structure I/145-BinaryTreePostOrderTraversal.java
/*
145. Binary Tree Postorder Traversal

Given the root of a binary tree, return the postorder traversal of its nodes' values.
Example:
    Input: root = [1,null,2,3]
    Output: [3,2,1] 

*/
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> oList = new ArrayList<Integer>();
        
        TraverseTree(root, oList);
        
        return oList;
    }
    
    public void TraverseTree(TreeNode node, List<Integer> oList) {
        if (node == null) {
            return;
        }
        
        TraverseTree(node.left, oList);
        TraverseTree(node.right, oList);
        
        oList.add(node.val);
    }
    
}